tango_simlib/tango_yaml_tools/main.py

```diff
@@ -75,8 +111,45 @@ def main():
         ),
     )
 
+    validate_parser = subparsers.add_parser(
+        "validate",
+        help=(
+            "Check conformance of a Tango device against a specification"
+            " in YAML format"
+        ),
+    )
+    validate_parser.add_argument(
+        "tango_device_name",
+        type=str,
+        help=(
+            "Tango device name in the format domain/family/member. "
+            "TANGO_HOST env variable has to be set"
+        ),
+    )
+    source_group = validate_parser.add_mutually_exclusive_group(required=True)
+    source_group.add_argument(
+        "-url", type=str, help="The URL to a YAML specification file",
+    )
+    source_group.add_argument(
+        "-path", type=str, help="The file path to a YAML specification file",
+    )
+
+    validate_parser.add_argument(
+        "-bidirectional",
+        action="store_true",
+        help=(
+            "When bidirectional is included, any details on the "
+            "device that is not in the spec is also listed."
+        ),
+    )
+
     args = parser.parse_args()
-    result = _build_yaml(args)
+
+    result = ""
+    if "url" in args:
+        result = _validate_device(args)
+    else:
+        result = _build_yaml(args)
     if result:
         print(result)
```

Comment:
For scripting, it would be useful to have a non-zero exit code if there are any differences.